crates/core/src/error.rs

```diff
@@ -34,10 +34,10 @@ pub enum Error {
         path: String,
     },
 
-    /// [geoarrow::error::GeoArrowError]
+    /// [geoarrow_array::error::GeoArrowError]
     #[error(transparent)]
     #[cfg(feature = "geoarrow")]
-    GeoArrow(#[from] geoarrow::error::GeoArrowError),
+    GeoArrow(#[from] geoarrow_array::error::GeoArrowError),
```

Comment:
I'm not exactly sure where `GeoArrowError` will live long term. `ArrowError` lives in `arrow_schema`, so maybe `GeoArrowError` should live in `geoarrow-schema`. But this means that it doesn't have access to the dependencies that other crates use.